include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -82,6 +82,170 @@ namespace internal
     {}
 
 
+
+    namespace
+    {
+      template <int dim, int spacedim>
+      unsigned int
+      setup_parallel_data_base_internal(
+        const unsigned int                                  n_subdivisions,
+        const dealii::hp::MappingCollection<dim, spacedim> &mapping_collection,
+        const std::vector<
+          std::shared_ptr<dealii::hp::FECollection<dim, spacedim>>>
+          &               finite_elements,
+        const UpdateFlags update_flags,
+        const bool        use_face_values,
+        std::vector<std::shared_ptr<dealii::hp::FEValues<dim, spacedim>>>
+          &x_fe_values,
+        std::vector<std::shared_ptr<dealii::hp::FEFaceValues<dim, spacedim>>>
+          &x_fe_face_values)
+      {
+        unsigned int n_q_points = 0;
+        if (use_face_values == false)
+          {
+            // determine if specific quadrature rules need to set up
+            bool needs_hypercube_setup = false;
+            bool needs_simplex_setup   = false;
+            bool needs_wedge_setup     = false;
+
+            for (const auto &fe : finite_elements)
+              for (unsigned int i = 0; i < fe->size(); ++i)
+                switch ((*fe)[i].reference_cell_type())
+                  {
+                    case ReferenceCell::Type::Vertex:
+                    case ReferenceCell::Type::Line:
+                    case ReferenceCell::Type::Quad:
+                    case ReferenceCell::Type::Hex:
+                      needs_hypercube_setup |= true;
+                      break;
+                    case ReferenceCell::Type::Tri:
+                    case ReferenceCell::Type::Tet:
+                      needs_simplex_setup |= true;
+                      break;
+                    case ReferenceCell::Type::Wedge:
+                      needs_wedge_setup |= true;
+                      break;
+                    default:
+                      Assert(false, ExcNotImplemented());
+                  }
+
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_simplex;
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_hypercube;
+            std::unique_ptr<dealii::Quadrature<dim>> quadrature_wedge;
+
+            if (needs_simplex_setup)
+              {
+                Assert(1 <= n_subdivisions && n_subdivisions <= 2,
+                       ExcNotImplemented());
+                quadrature_simplex = std::make_unique<Quadrature<dim>>(
```

Comment:
Why do you need the assertion? Because you currently don't have `Simplex::FE_P` implemented for higher polynomial degrees? That's something that `FE_P` should assert, not here